junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/IsPotentialJUnit4TestClass.java

```diff
@@ -22,12 +23,20 @@
 
 	@Override
 	public boolean test(Class<?> candidate) {
-		// Do not collapse into single return.
 		if (isAbstract(candidate))
 			return false;
 		if (!isPublic(candidate))
 			return false;
-		return !candidate.isMemberClass();
+		// Do not collapse into single return.
```

Comment:
So the warning moved here.
![warning-moved2](https://cloud.githubusercontent.com/assets/5991375/24840204/350c6242-1d69-11e7-80de-161d13ee4170.png)

So I should keep the comment here?